xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -330,6 +330,9 @@ private void handleRpcStreamClosed(Status error) {
         return;
       }
 
+      // Need this here to avoid tsan race condition in XdsClientImplTestBase.sendToNonexistentHost
+      long elapsed = stopwatch.elapsed(TimeUnit.NANOSECONDS);
```

Comment:
This really needs to be together with actually scheduling the callback, as lots of time can pass within `handleStreamClosed()`. Move the rest of that lower code involving rescheduling up here (mainly before handleStreamClosed()).

Reply:
handleStreamClosed() causes retryBackoffPolicy.nextBackoffNanos() to change, so moving the rest of the rescheduling up causes 10 tests to fail with missing interactions.

What about just making FakeClock.currentTimeNanos volatile?

Reply:
> handleStreamClosed() causes retryBackoffPolicy.nextBackoffNanos() to change, so moving the rest of the rescheduling up causes 10 tests to fail with missing interactions.

Really? I tried multiple options of code placement and couldn't get xds tests to fail.

`volatile` doesn't seem to help. `sendToNonexistentHost()` explicitly says it did `forwardTime()` to trigger "rpcRetry". Isn't that talking about this code here? If time is racy (by 50 seconds!), then the `schedule()` will be wildly off, or not having actually executed yet.

Reply:
NVM, the problem was that retryBackoffPolicy was null because I hadn't moved the block that initializes it.  Once I moved the block and got rid of the NPEs, then everything worked fine.